Reset GUI

superstar54 opened the pull request:
Reset the GUI when selecting to start a new workflow. Related to  #437 #420. 

Reset:
- step 1, Structure (structure and viewer)
- step 2, Configuration (all panels)
- step 3, computational resource. (code and resources)

Step 4 will be automatically reset when the `process` of step 3 is reset.

AndresOrtegaGuerrero commented:
@superstar54  I did a test, and it seems the Smearing type: , Smearing width (Ry) arent reset , also Wavefunction cutoff (Ry)
Charge density cutoff (Ry) . I am also wondering if the options in the panels get also reset ? I guess each panel needs a method named reset ?

superstar54 commented:
Thanks for the review. 

>  it seems the Smearing type: , Smearing width (Ry) arent reset , also Wavefunction cutoff (Ry)
Charge density cutoff (Ry) . 

I fixed it. Also fixed the `reset` in `pseudo_setter` and `magnetization`.

> I am also wondering if the options in the panels get also reset ? I guess each panel needs a method named reset ?

yes, each panel needs to implement a `reset` method, which will be called from the `ConfigureQeAppWorkChainStep`.

AndresOrtegaGuerrero commented:
@superstar54 Would it be necessary to add in the documentation of qe_app in `create your plugin` , to include a reset method, to avoid future conflicts ?

AndresOrtegaGuerrero commented:
@superstar54 I tested the PR, now it does update the field i told you , yet I noticed some behavior. I started to select one workchain and then another , and then the first one again. (Everything was working) after another change the app crashed.
![Screenshot 2023-10-10 at 09 47 23](https://github.com/aiidalab/aiidalab-qe/assets/34098967/4a00f4ff-0ad5-4e86-8df2-acb68834a437)

AndresOrtegaGuerrero commented:
@superstar54 another thing i noticed is that one of my workchain , (newer one) doesnt set the input parameters, but the another one i was using for testing it does. I know this is related to Load GUI PR.

AndresOrtegaGuerrero commented:
>Which workchain? In my test, the load_GUI is OK.

It seems my workchain is "old" so i tried with a new one an no issues , maybe that workchain is incompatible with the load_GUI, but indeed i tried a new one and everything upload correctly

AndresOrtegaGuerrero commented on src/aiidalab_qe/app/configuration/pseudos.py:
```diff
@@ -278,7 +279,7 @@ def _on_cutoff_change(self, _=None):
+        self.ecutwfc_setter.value = 0
         self.ecutrho_setter.value = 0
 
     def _reset_traitlets(self):
```
Maybe you should include docstring to these _reset_* functions , just to keep it consistent with the rest of your changes where you added docstring

superstar54 commented:
Hi @AndresOrtegaGuerrero , thanks for the review. I made the changes as you requested.
- add docstring to the `reset` methods.
- remove the message of `Stored in AiiDA....`

AndresOrtegaGuerrero reviewed (approved):
Thank you! LGTM. One note once you use .reset() for the settings in the panels, if a future plugin doesnt have this method it should be a problem right? I think that just be a different issue for the documentation for plugin developers

superstar54 commented:
> One note once you use .reset() for the settings in the panels, if a future plugin doesnt have this method it should be a problem right? I think that just be a different issue for the documentation for plugin developers

Thanks for the comment. In the `Panel` class, there is an empty `reset` method to ensure it always exists.

unkcpz commented:
Thanks @superstar54. This is finally happened, great!

> In the Panel class, there is an empty reset method to ensure it always exists.

However, there is no guarantee that user (property plugin developers) will implement this. I think if the `reset` is the method that must be overridden otherwise the app is not working properly, you need make the `reset` of `Panel` raise `NotImplement` exception or provide the default behaviour to avoid breaking the app. 
If this is the case, please open an issue for future implementation. Thanks!
